sdk/cosmos/azure-cosmos/src/main/java/com/azure/cosmos/patch/PatchOperationCore.java

```diff
@@ -0,0 +1,32 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+package com.azure.cosmos.patch;
+
+import com.azure.cosmos.implementation.apachecommons.lang.StringUtils;
+import com.azure.cosmos.patch.implementation.PatchOperationType;
+
+import static com.azure.cosmos.implementation.guava25.base.Preconditions.checkArgument;
+
+public class PatchOperationCore<TResource>  extends PatchOperation<TResource> {
```

Comment:
shouldn't be public?

Reply:
It's needed as I am accessing it in another package. But this is in implementation package. So should be fine. I have made it final class so no one can extent it.